packages/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,301 @@
+  variant = Variant.Light,
+  triggerEvent = TriggerEvent.Hover,
+  align = 'top',
+  justify = 'start',
+  id,
+  shouldClose,
+  ...rest
+}: TooltipProps) {
+  const isControlled = !!controlledSetOpen;
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState(false);
+  const open = isControlled ? controlledOpen : uncontrolledOpen;
+  const setOpen = isControlled ? controlledSetOpen : uncontrolledSetOpen;
+
+  const triggerRef = useRef<HTMLElement>(null);
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  const tooltipId =
+    id || `tooltip-${Math.floor(Math.random() * Math.floor(10))}`;
+
+  const triggerEventMap = (triggerType: TriggerEvent, triggerProps?: any) => {
+    if (triggerType === 'hover') {
+      return {
+        onMouseEnter: debounce(() => {
+          setOpen && setOpen(!open);
+        }, 250),
+        onMouseLeave: debounce(() => {
+          handleClose();
+        }, 250),
+        onFocus: () => setOpen && setOpen(true),
+        onBlur: handleClose,
+      };
+    } else {
+      if (triggerProps.onClick) {
+        return {
+          onClick: () => {
+            triggerProps.onClick();
+            setOpen && setOpen(!open);
+          },
+        };
+      } else {
+        return {
+          onClick: () => setOpen && setOpen(!open),
+        };
+      }
+    }
+  };
+
+  const handleClose = () => {
+    if (!shouldClose || shouldClose()) {
+      setOpen && setOpen(false);
+    }
+  };
+
+  const handleEscape = (e: KeyboardEvent) => {
+    e.stopImmediatePropagation();
+
+    if (e.keyCode === escapeKeyCode) {
+      handleClose();
+    }
+  };
+
+  useEventListener('keydown', handleEscape);
+
+  const handleBackdropClick = (e: MouseEvent) => {
+    const tooltipReference = tooltipRef && tooltipRef.current;
+
+    if (
+      triggerEvent === 'click' &&
+      tooltipReference &&
+      !tooltipReference.contains(e.target as HTMLElement)
+    ) {
+      handleClose();
+    }
+  };
+
+  const enabled = open;
+
+  useEventListener('click', handleBackdropClick, { enabled });
+
+  const triggerRect =
+    triggerRef &&
+    triggerRef.current &&
+    triggerRef.current.getBoundingClientRect();
+
+  // We are doing this to get the final alignment and justification from Popover
+  // And to make sure we're responding to not just the desired alignment, but the alignment
+  // As a result of our calcPosition() function
+  const [alignment, setAlignment] = useState(align);
+  const [justification, setJustification] = useState(justify);
+
+  const triangleStyle = trianglePosition(
+    alignment,
+    justification,
+    triggerRect,
+  ) as any;
+
+  const tooltip = (
+    <Popover
+      key="popover"
+      active={open}
+      align={align}
+      justify={justify}
+      usePortal={true}
+      adjustOnMutation={true}
+      spacing={12}
+      setAlignment={setAlignment}
+      setJustification={setJustification as any}
+    >
+      <div
+        {...rest}
+        role="tooltip"
+        id={tooltipId}
```

Comment:
What's this being used for?

Reply:
Accessibility. Matches `aria-describedby`